app/controllers/admissions_admin/admissions_controller.rb

```diff
@@ -99,6 +231,37 @@ def admission_params
 
 private
 
+  # Calculate who should receive rejection email
+  # Excludes those who already received an email
+  def calculate_rejection_recipients
+    rejected = automatically_rejected_applicants
+    already_sent = @admission.rejection_emails.map { |r| r.applicant_id }
+    # Select all not already sent
+    rejected.select { |r| not already_sent.include?(r.id) }
+  end
+
+  # Calculates who are marked to receive automatic rejection email
+  def automatically_rejected_applicants
+    rejected_somewhere = []
+    accepted_somewhere = []
+
+    Group.all.each do |g|
+      jobs = g.jobs_in_admission(@admission)
+      jobs.each do |j|
+        rejected_somewhere += j.automatically_rejected_applications
+        accepted_somewhere += j.accepted_applications
+      end
+    end
+
+    # Convert to unique applicants
+    rejected_somewhere = rejected_somewhere.flatten.map { |x| x.applicant }.uniq
+    accepted_somewhere = accepted_somewhere.flatten.map { |x| x.applicant }.uniq
+
+    # Return those rejected somewhere if not accepted anywhere
+    rejected_somewhere - accepted_somewhere
+  end
+
```

Comment:
This function and "calculate_rejection_recipients" are very important to verify for correctness!